src/OrleansAzureUtils/Storage/AzureQueueDataManager.cs

```diff
@@ -140,12 +139,13 @@ public AzureQueueDataManager(string queueName, string deploymentId, string stora
             {
                 // that way we don't have first to create the queue to be able later to delete it.
                 CloudQueue queueRef = queue ?? queueOperationsClient.GetQueueReference(QueueName);
-                var exists = Task<bool>.Factory.FromAsync(queueRef.BeginExists, queueRef.EndExists, null);
+                var exists = queueRef.ExistsAsync();
                 if (await exists)
```

Comment:
Doesn't matter, but this could be inlined without losing clarity, or the await could be moved up.